File: backend/src/main/java/com/sentinel/backend/entity/PlanoPagamento.java
package com.sentinel.backend.entity;

import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "planos_pagamento")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PlanoPagamento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String descricao;
    private int numeroParcelas;
    private String periodicidade;
    private BigDecimal valorTotal;
}

File: frontend/src/app/models/plano-pagamento.ts
export class PlanoPagamento {
  id?: number;
  descricao: string;
  numeroParcelas: number;
  periodicidade: string;
  valorTotal: number;

  constructor(
    descricao: string,
    numeroParcelas: number,
    periodicidade: string,
    valorTotal: number,
    id?: number
  ) {
    this.id = id;
    this.descricao = descricao;
    this.numeroParcelas = numeroParcelas;
    this.periodicidade = periodicidade;
    this.valorTotal = valorTotal;
  }
}
